### PR TITLE
Create dvc.yaml file with training script on run experiment

### PR DESCRIPTION
**Pull request by sroy3**
Part of #3058 

https://user-images.githubusercontent.com/3683420/216372454-7f4f86ea-67c4-4a88-b9bd-4a63cedb0479.mov

### CODE REVIEW

**Comment by shcheklein** [comment]
Thanks @sroy3 !

Two things to change/improve.

- we want path to be relative to the project root in the `dvc.yaml`
- we don't need `out` `dvclive` since we are not 100% sure if that it the case or we might not know the path (it can be customized)

we can discuss as a next step what to include in the file

**Comment by shcheklein** [comment]
We'll need to handle the case if path is a notebook ... (again, can be a followup) ... or a non-Python command ...

**Comment by sroy3** [comment]
> Thanks @sroy3 !
> 
> Two things to change/improve.
> 
> * we want path to be relative to the project root in the `dvc.yaml`
> * we don't need `out` `dvclive` since we are not 100% sure if that it the case or we might not know the path (it can be customized)
> 
> we can discuss as a next step what to include in the file

Done.

**Comment by sroy3** [comment]
Updated demo:

https://user-images.githubusercontent.com/3683420/216689309-a9da898a-375c-4239-b60b-5609b8c68b66.mov

**Comment by shcheklein** [comment]
@sroy3 thanks for the update!

Q - could you please remind me what is the current logic to show the table (when there are no columns) and when we show the Settings page instead.

**Comment by sroy3** [comment]
> @sroy3 thanks for the update!
> 
> Q - could you please remind me what is the current logic to show the table (when there are no columns) and when we show the Settings page instead.

That's it. I may have cheated in my demo using the beaker instead of the "Show Experiments" button, which did always show the table, but that button isn't there anymore.

**Comment by sroy3** [comment]
> We'll need to handle the case if path is a notebook ... (again, can be a followup) ... or a non-Python command ...

Added a [comment](https://github.com/iterative/vscode-dvc/issues/3058#issuecomment-1419516895) for that and currently working on this.

**Comment by shcheklein** [comment]
>That's it. I may have cheated in my demo using the beaker instead of the "Show Experiments" button, which did always show the table, but that button isn't there anymore.

oh, got it. I wasn't aware about this backdoor :) There none left, right?

> Updated demo:

Perfect, if everything else looks good, we can merge this, I think.

File: extension/src/experiments/workspace.test.ts [hunk 1]
```diff
@@ -244,4 +254,131 @@ describe('Experiments', () => {
...
+
+      await workspaceExperiments.getCwdThenRun(mockedCommandId, true)
+
+      expect(executeCommandSpy).toHaveBeenCalledWith(
+        mockedCommandId,
+        mockedDvcRoot
+      )
+    })
+
+    it('should not run the command if the path to the taining script was not given', async () => {
```
Comment (julieg18) [inline extension/src/experiments/workspace.test.ts:366]:
```suggestion
    it('should not run the command if the path to the training script was not given', async () => {
```

File: extension/src/experiments/workspace.ts [hunk 2]
```diff
@@ -429,6 +440,36 @@ export class WorkspaceExperiments extends BaseWorkspaceWebviews<
...
 
+  private async checkOrAddPipeline(cwd: string) {
+    const stages = await this.internalCommands.executeCommand(
+      AvailableCommands.STAGE_LIST,
+      cwd
+    )
+
+    if (!stages) {
+      const selectValue = 'select'
+      const pathOrSelect = await quickPickOneOrInput(
```
Comment (julieg18) [inline extension/src/experiments/workspace.ts:451]: Should we put down some kind of Toast message mentioning we're creating a dvc.yaml file for running experiments?

File: extension/src/experiments/workspace.ts [hunk 3]
```diff
@@ -429,6 +440,36 @@ export class WorkspaceExperiments extends BaseWorkspaceWebviews<
...
+        [{ label: 'Select from file explorer', value: selectValue }],
+        {
+          defaultValue: '',
+          placeholder: 'Path to script',
```
Comment (julieg18) [inline extension/src/experiments/workspace.ts:455]: We should probably be more clear that only absolute paths work, not relative. Personally, I think that we should allow relative paths in the input if possible. Seems easier to type out 🤔

**Review by julieg18 — APPROVED** [review]
Great work!

**Comment by sroy3** [comment]
> > That's it. I may have cheated in my demo using the beaker instead of the "Show Experiments" button, which did always show the table, but that button isn't there anymore.
> 
> oh, got it. I wasn't aware about this backdoor :) There none left, right?

Right!

File: extension/src/experiments/workspace.ts (hunk 2 above)
Comment (sroy3) [reply extension/src/experiments/workspace.ts:451]: I'm going to open the `dvc.yaml` file automatically in a follow-up.

**Comment by mattseddon** [comment]
> > > That's it. I may have cheated in my demo using the beaker instead of the "Show Experiments" button, which did always show the table, but that button isn't there anymore.
> > 
> > 
> > oh, got it. I wasn't aware about this backdoor :) There none left, right?
> 
> Right!

Can you get to the table from `DVC: Show Experiments` in the command palette?

**Comment by sroy3** [comment]
> > > > That's it. I may have cheated in my demo using the beaker instead of the "Show Experiments" button, which did always show the table, but that button isn't there anymore.
> > > 
> > > 
> > > oh, got it. I wasn't aware about this backdoor :) There none left, right?
> > 
> > 
> > Right!
> 
> Can you get to the table from `DVC: Show Experiments` in the command palette?

You are right. Would it be better to hide the show commands until the project is ready or to redirect to the setup?

**Comment by mattseddon** [comment]
> You are right. Would it be better to hide the show commands until the project is ready or to redirect to the setup?

Hiding is probably easier but up to you. Will need to go through the same process for plots.

**Comment by shcheklein** [comment]
Let's not hide it please. Ultimately, with the onboarding we should be trying to make the product look and feel 100% the same if something is installed or not. And we should be helping as we go collecting pieces of the puzzle, ideally through the existing interface as much as possible.

In this sense, I'm considering even the existence of the Setup page as a compromise ;)

I think the jupyter extension experience is a good example.

**Pull request merged**